workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.ts

```diff
@@ -43,127 +58,148 @@ export class DynamicAuthorizationService {
     user: AuthenticatedUser,
     isAuthorizedOnSubjectRequest: IsAuthorizedOnSubjectRequest
   ): Promise<void> {
-    throw new Error('NotImplemented');
+    const { id, roles } = user;
+    const { subjectId, subjectType, action } = isAuthorizedOnSubjectRequest.dynamicOperation;
+    const { groupIds } = await this._dynamicPermissisonsPlugin.getUserGroups({
+      userId: id
+    });
+    const groupIdIdentities = Array.from(new Set<string>([...groupIds, ...roles])).map(
+      (groupId): { identityType: IdentityType; identityId: string } => {
```

Comment:
`{ identityType: IdentityType; identityId: string }` has been used in a couple places. Might be worth it to extract to an interface